internal/commands/sbomtest/snykclient/snykclient.go

```diff
@@ -0,0 +1,126 @@
+package snykclient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+const sbomTestAPIVersion = "2023-08-31~beta"
+
+type SnykClient struct {
+	http       *http.Client
+	apiBaseURL string
+	orgID      string
+}
+
+func NewSnykClient(httpClient *http.Client, apiBaseURL, orgID string) *SnykClient {
+	return &SnykClient{
+		http:       httpClient,
+		apiBaseURL: apiBaseURL,
+		orgID:      orgID,
+	}
+}
+
+func (s *SnykClient) CreateSBOMTest(ctx context.Context, sbomReader io.Reader) (*SBOMTest, error) {
+	url := fmt.Sprintf("%s/rest/orgs/%s/sbom_tests?version=%s", s.apiBaseURL, s.orgID, sbomTestAPIVersion)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, sbomReader)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/vnd.api+json")
+
+	rsp, err := s.http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var response CreateSBOMTestRunResponseBody
+	if err := parseResponse(rsp, http.StatusCreated, &response); err != nil {
+		return nil, err
+	}
+
+	return &SBOMTest{
+		Client: s,
+		ID:     response.Data.ID,
+	}, nil
+}
+
+// SBOM Test Client
+
+type SBOMTest struct {
+	Client *SnykClient
+	ID     string
+}
```

Comment:
should we decouple `SBOMTest` from `SnykClient`?

```suggestion
type SBOMTest struct {
	client *http.Client
	ID     string
	orgID  string
}
```